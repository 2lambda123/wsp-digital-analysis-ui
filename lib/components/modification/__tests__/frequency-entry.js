/* global describe, it, expect, jest */

describe('Component > FrequencyEntry', () => {
  const renderer = require('react-test-renderer')
  const React = require('react')
  const {
    mockFeed,
    mockPattern,
    mockTimetable
  } = require('../../../utils/mock-data')
  const FrequencyEntry = require('../frequency-entry')

  it('renders correctly', () => {
    const replaceTimetableFn = jest.fn()
    const removeTimetableFn = jest.fn()
    const setActiveTripsFn = jest.fn()
    const tree = renderer
      .create(
        <FrequencyEntry
          allPhaseFromTimetableStops={{}}
          entry={mockTimetable}
          feed={mockFeed}
          modificationStops={[]}
          update={replaceTimetableFn}
          remove={removeTimetableFn}
          projectTimetables={[]}
          setActiveTrips={setActiveTripsFn}
          routes={['route1']}
          routePatterns={[mockPattern]}
          trip='abcd'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    expect(replaceTimetableFn).not.toBeCalled()
    expect(removeTimetableFn).not.toBeCalled()
    expect(setActiveTripsFn).not.toBeCalled()
  })
})
