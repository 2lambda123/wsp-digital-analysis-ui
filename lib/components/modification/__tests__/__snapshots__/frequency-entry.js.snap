// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > FrequencyEntry renders correctly 1`] = `
<section
  className="panel panel-default inner-panel"
  onFocus={[Function]}
>
  <a
    className="panel-heading clearfix"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
    tabIndex={0}
  >
    <Icon
      type="calendar"
    />
    <strong>
       
      Test timetable
    </strong>
    <Icon
      className="pull-right"
      type="caret-right"
    />
  </a>
</section>
`;
